engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/job/preparation/JobPreparationActor.scala

```diff
@@ -149,10 +149,14 @@ class JobPreparationActor(workflowDescriptor: EngineWorkflowDescriptor,
       case Success(_: DockerImageIdentifierWithoutHash) if !hasDockerDefinition => 
         lookupKvsOrBuildDescriptorAndStop(inputs, attributes, NoDocker, None)
 
-      // Even if the docker image has a hash, we need to (try to) find out the size, so send a request
-      case Success(dockerImageId: DockerImageIdentifierWithHash) =>
+      // If there's a digest and performRegistryLookupIfDigestIsProvided, look up the container
+      case Success(dockerImageId: DockerImageIdentifierWithHash) if DockerConfiguration.instance.performRegistryLookupIfDigestIsProvided =>
```

Comment:
Just a small comment, I'd probably move the `if` inside the `case Success` to make it clearer that these are two alternative ways of handling the case of `Success(dockerImageId: DockerImageIdentifierWithHash)`, and to avoid having to perform the same pattern match twice

Reply:
Ah yep perfect, I've refactored this now.

```scala
case Success(dockerImageId: DockerImageIdentifierWithHash) =>
  // If there's a digest and:
  if (DockerConfiguration.instance.performRegistryLookupIfDigestIsProvided) {
    // performRegistryLookupIfDigestIsProvided is true (default), look up the container
    sendDockerRequest(dockerImageId)
  } else {
    // performRegistryLookupIfDigestIsProvided is false, we'll skip looking up the container
    lookupKvsOrBuildDescriptorAndStop(inputs, attributes, DockerWithHash(dockerImageId.fullName), None)
  }
```